src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -436,6 +436,21 @@ protected override int MakeSlot(BoundExpression node)
                         }
                     }
                     break;
+                case BoundKind.DefaultExpression:
+                    {
+                        var type = ((BoundDefaultExpression)node).Type;
+                        if (EmptyStructTypeCache.IsTrackableStructType(type))
+                        {
+                            int slot = GetOrCreateObjectCreationPlaceholderSlot(node);
```

Comment:
>int slot = GetOrCreateObjectCreationPlaceholderSlot(node); [](start = 28, length = 58)

It feels like there is an inconsistency between when we create placeholder locals for the ```default``` expression and for an object creation. I think it would be better to use the same approach for all such cases. For example, here we would simply do what is done for the ```case BoundKind.ObjectCreationExpression:``` and the code that is currently here could be moved to ```VisitDefault``` etc.

Reply:
Yes, that is necessary to handle split states as well.

---
In reply to: [246945160](https://github.com/dotnet/roslyn/pull/32304#discussion_r246945160) [](ancestors = 246945160)